inc/cloudfront_media_purge/namespace.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Altis\Cloud\Cloudfront_Media_Purge;
+
+use Exception;
+use function Altis\get_aws_sdk;
+
+function bootstrap() {
+	add_action( 'delete_attachment', __NAMESPACE__ . '\\purge_media_file_cache', 100 );
+}
+
+/**
+ * Return an AWS CloudFront Client instance
+ *
+ * @return \Aws\CloudFront\CloudFrontClient
+ */
+function get_aws_client() {
+	return get_aws_sdk()->createCloudFront( [
+		'version' => 'latest',
+	] );
+}
+
+/**
+ * Purge an attachment file cache from CloudFront
+ *
+ * It sends a request to invalidate all sizes given for a media file
+ *
+ * @param int $media_id Attachment ID.
+ *
+ * @return bool
+ */
+function purge_media_file_cache( $media_id ) {
+	$meta = wp_get_attachment_metadata( $media_id );
+	if ( ! $meta ) {
+		return false;
+	}
+	$baseurl = '/tachyon/';
+	$basedir = trailingslashit( dirname( $meta['file'] ) );
+	$items   = [ $baseurl . $meta['file'] . '*' ];
+
+	if ( ! empty( $meta['sizes'] ) ) {
+		foreach ( $meta['sizes'] as $size => $size_meta ) {
+			$items[] = $baseurl . $basedir . $size_meta['file'] . '*';
+		}
+	}
+
+	$client = get_aws_client();
+
+	$distribution_id = apply_filters( 'altis_cloudfront_media_purge_distribution_id', defined( 'CLOUDFRONT_DISTRIBUTION_ID' ) ? CLOUDFRONT_DISTRIBUTION_ID : '' );
```

Comment:
It's good to try and avoid ternaries if possible, they're less readable even if they're more terse.

I'd do something like:

```php
$distribution_id = '';
if ( defined( 'CLOUDFRONT_DISTRIBUTION_ID' ) ) {
    $distribution_id = CLOUDFRONT_DISTRIBUTION_ID;
}
```

You should also add doc blocks to filters.

Lastly we use dots in our filter names for namespaces - have a look at some other examples in the file, so might be like `altis.cloud.media_purge_distribution_id`